modules/user_accounts/ajax/rejectUser.php

```diff
@@ -14,63 +19,44 @@
  * @link     https://github.com/aces/Loris
  */
 
-use LORIS\user_accounts as UA;
+namespace LORIS\user_accounts;
 
-define('NO_IDENTIFIER_SUPPLIED', 1);
-define('INCORRECT_PERMISSION', 2);
-define('ACCOUNT_ACTIVE', 3);
+// Make sure the permission is checked first to avoid malicious userID scans
+if (!(\User::singleton())->hasPermission('user_accounts')) {
+    header("HTTP/1.1 403 Forbidden");
+    print_r(
+        "You do not have the correct permissions for this 
+         operation (need admin or user accounts)"
+    );
+    exit(1);
+}
 
-if (isset($_POST['identifier'])) {
-    $identifier = $_POST['identifier'];
-    _rejectUser($identifier);
-} else {
-    throw new LorisException("No identifier supplied", NO_IDENTIFIER_SUPPLIED);
+if (!isset($_POST['identifier'])) {
+    header("HTTP/1.1 400 Bad Request");
+    print_r(
+        "No identifier supplied"
+    );
+    exit(2);
 }
 
-/**
- * Checks that logged in user has user_accounts permissions, which is
- * also included in admin permissions
- *
- * @return boolean true if user has user accounts permissions, false otherwise
- */
-function _hasPerm()
-{
-    $user =& User::singleton();
-    if (isset($user)) {
-        return $user->hasPermission('user_accounts');
-    }
-    return false;
+$username = (\User::factory($_POST['identifier']))->getUsername();
+
+if (empty($username)) {
+    header("HTTP/1.1 404 Not Found");
+    print_r(
+        "This account do not exists"
```

Comment:
Everything looks great except for this little typo, which i can fix. thanks @xlecours, the code is very pretty :)